pkg/lockfile/dpkg-status.go

```diff
@@ -122,7 +124,7 @@ func (e DpkgStatusExtractor) Extract(f DepFile) ([]PackageDetails, error) {
 
 		// PackageDetails does not contain any field that represent a "not installed" state
 		// To manage this state and avoid false positives, empty struct means "not installed" so skip it
-		if (PackageDetails{}) == pkg {
+		if reflect.DeepEqual(PackageDetails{}, pkg) {
```

Comment:
my understanding is that pulling in `reflect` can be a big deal in Go, and at at least in the past with some methods (which tbh probably don't include `DeepEqual`) could trigger sizeable deoptimizations.

I don't think this check is anything special and very internal so could be easily replaced with something like returning a pointer (so that we can then return `nil`), returning an `error` that we can eat here, or setting the `Name` to something like `<not installed>`.

I think that's probably worth doing either way so that we're not as dependent on the _whole_ structure of `PackageDetails`

Reply:
I modified the code here to check each field in `PackageDetails` and if they are all "empty". wdty?